java/test/src/main/java/org/ray/api/test/RayCallTest.java

```diff
@@ -85,9 +88,12 @@ public void testType() {
     LargeObject largeObject = new LargeObject();
     Assert.assertNotNull(Ray.call(RayCallTest::testLargeObject, largeObject).get());
 
-    ObjectId randomObjectId = ObjectId.fromRandom();
-    Ray.call(RayCallTest::testNoReturn, randomObjectId);
-    Assert.assertEquals(((int) Ray.get(randomObjectId)), 1);
+    // TODO(edoakes): this test doesn't work now that we've switched to direct call
```

Comment:
cc @kfstorm

Reply:
I don't get it. We've dropped non-direct call support in Java via https://github.com/ray-project/ray/pull/7408. So this case should work in direct call mode. Why the removing of the transport type flag impacts this test case?

Also, can you point out what the protocol is?

Reply:
I assume this was working because the object was being put in plasma by the remote task and then the driver fetches it via plasma. This no longer works because we always check the in-memory store before fetching an object. In order for this to work, the driver needs to contact the owner (the remote task in this case) first.

Here is where it's implemented in python:
https://github.com/ray-project/ray/blob/master/python/ray/_raylet.pyx#L1068
https://github.com/ray-project/ray/blob/master/python/ray/serialization.py#L159

Reply:
I see. The current way to detect whether the no-return task is executed is too tricky. It involves a private API to put an object with a specified object ID. I think we can modify it to write a temp file in the disk and check that in the driver. What do you think?